feat(settings): setup preferred weight units

mitchelljoram opened the pull request:
- Added preferred weight unit in the user model and entity, and updated user repository.
- Added weight unit preset to the setting's page (need to update UI).
    - Added weight unit controller to update user model and entity.
- Updated live tracker to take in preset weight unit on the summary bar and summary page

mitchelljoram committed:
Added preferred weight unit to user

feat: added weight unit enum and added preferred weight unit to user entity and model

mitchelljoram committed:
Added weight unit preset setting

feat: added weight unit preset to the settings page that updates user's preferred weight unit

mitchelljoram committed:
Added user weight unit preset to live tracker

feat!: Updated live tracker to utilize user's weight unit preset

MattyD1 commented:
Just for future reference, make sure you use the Pull Request template.

MattyD1 commented on lib/features/auth/data/models/user_model.dart:
```diff
@@ -14,6 +14,7 @@ class UserModel with _$UserModel {
     required String email,
     required bool isMale,
     required ThemeMode preferredTheme,
+    required WeightUnit preferredWeightUnit,
```
This enum is never declared in the PR

mitchelljoram replied:
changed this enum to the Units enum that was already present

MattyD1 commented on lib/features/auth/data/repositories/user_repository.dart:
```diff
@@ -81,4 +81,27 @@ class UserRepository {
+      final user = res.first;
+
+      box.put(
+        user..dbPreferredWeightUnit = unit.index,
```
You don't have to use the dbPreferredWeightUnit here, you can use the transient preferredWeightUnit instead.

This saves you from doing the index call.

mitchelljoram replied:
updated user preferred theme and weight units to use the enums

MattyD1 commented on lib/features/system/ui/components/weight_unit_display.dart:
I dont understand why we have this file since you have the weight_unit_selector container that handles the state management call.  I think we can get rid of this.

MattyD1 commented on lib/features/tracker/ui/containers/workout_summary_header.dart:
```diff
@@ -13,9 +14,10 @@ class WorkoutSummaryHeader extends ConsumerWidget {
     final totalVolume = ref
         .watch(liveWorkoutControllerProvider.notifier)
-        .getTotalVolume(Units.lbs);
+        .getTotalVolume(Units.values[weightUnit.index]);
```
We can just pass in the Units enum here instead of using the index.

Also why did you create a secondary WeightUnits enum instead of using the existing Units enum?

mitchelljoram replied:
Changed to strickly use Units enum

MattyD1 commented on lib/features/tracker/ui/containers/main_tracker_summary.dart:
```diff
@@ -54,9 +55,10 @@ class _MainTrackerSummaryState extends ConsumerState<MainTrackerSummary> {
     final totalVolume = ref
         .watch(liveWorkoutControllerProvider.notifier)
-        .getTotalVolume(Units.lbs);
+        .getTotalVolume(Units.values[weightUnit.index]);
```
We can just pass in the Units enum here instead of using the index.

MattyD1 commented on lib/features/tracker/ui/containers/workout_summary_header.dart:
```diff
@@ -43,7 +45,7 @@ class WorkoutSummaryHeader extends ConsumerWidget {
                 color: context.colors.blue,
                 value: totalVolume.cleanNumber(decimal: 0),
-                label: 'lbs',
+                label: Units.values[weightUnit.index].name,
```
We can just pass in the Units enum here instead of using the index.

MattyD1 commented on pubspec.lock:
Which version of the flutter sdk do you have I may need to upgrade to ensure consistency

mitchelljoram replied:
Flutter 3.24.3

MattyD1 reviewed (changes requested):
Make sure to update the UI, and make sure the tests are passing as well.

mitchelljoram committed:
Updated preferred weight units to use units enum

feat!: updated user model, user entity, and component preferred weight units to use units enum

mitchelljoram committed:
Added preferred weight unit to testing

feat(test): added preferred weight unit to testing

mitchelljoram committed:
Updated weight unit preset UI

feat(settings): updated weight unit preset UI to a radio list

mitchelljoram committed:
Removed unnecessary imports

feat: removed unnecessary imports

MattyD1 reviewed (changes requested):
Tests are still failing, so you will need to address those issues.

Also you did not address some of the comments I made in my last review, if you could take a look at them.

FInally I am not sure if I like the UI that much I don't think it fits with the rest of the screen.  This is a non-blocker but I think we can look into this a bit more and draft a figma design.

mitchelljoram committed:
Fixed trailing comma issues

fix: Fixed trailing comma issues

mitchelljoram committed:
feat(user): updated user repository to use enums instead of indexes

MattyD1 commented on lib/features/system/ui/components/weight_unit_display.dart:
I still think we should remove this file, and consolidate the code into weight_unit_selector.dart

MattyD1 reviewed (changes requested):
One last thing is I think now that the Weight Unit enum is gone we should rename weightUnits to just units.  We can always refactor is we choose to add time as a measurement later
